code_review.md

```diff
@@ -0,0 +1,98 @@
+
+# Purpose
+Code review is an important part of a team's development process. It helps to:
+
+* disseminate knowledge about the codebase/technology/techniques across teams
+* increase awareness of the features being developed
+* provide opportunities to educate junior engineers (actually all engineers)
+* find alternative solutions to problems
+* catch poor architectural decisions and bugs
+* develop a sense of team ownership of the code (no more his/her code)
+
+When doing code reviews, **the process is as important as the result**, therefore, some guidelines are presented below to help improve the process of code review.
+
+# Everyone
+* Accept that many programming decisions are opinions. Discuss tradeoffs, which
+you prefer, and reach a resolution quickly.
+* Ask questions; don't make demands. ("What do you think about naming this
+`:user_id`?")
+* Ask for clarification. ("I didn't understand. Can you clarify?")
+* Avoid selective ownership of code. ("mine", "not mine", "yours")
+* Avoid using terms that could be seen as referring to personal traits. ("dumb",
+"stupid"). Assume everyone is attractive, intelligent, and well-meaning.
+* Be explicit. Remember people don't always understand your intentions online.
+* Be humble. ("I'm not sure - let's look it up.")
+* Don't use hyperbole. ("always", "never", "endlessly", "nothing")
+* Don't use sarcasm.
+* Talk in person if there are too many "I didn't understand" or "Alternative
+solution:" comments. Post a follow-up comment summarizing offline discussion.
+
+# Having Your Code Reviewed
+* Review your code before anybody else.
+* Provide context about what the change is and why is it important - Write informative commit messages [4] and pull request descriptions.
+* Use the pull request description template [3]. You only need to answer the relevant questions. Pro tip: use the bookmarklet [11] or chrome extension [12] for pull request description templates.
+* Annotate your code reviews. Add comments anywhere that you feel will help the reviewer understand your code.
+* Be grateful for the reviewer's suggestions. ("Good call. I'll make that
+change.")
+* Don't take it personally. The review is about the code, not you.
+* Explain why the code exists. ("It's like that because of these reasons. Would
+it be more clear if I rename this class/file/method/variable?")
+* Seek to understand the reviewer's perspective.
+* Try to respond to every comment.
+* Wait to merge the branch until someone has :+1:'d your PR and Continuous Integration
+tells you the test suite is green in the branch.
+* Merge once you feel confident in the code and its impact on the project.
+* Delete the backing branch.
+
+# Reviewing Code
+* Code review often and for short sessions. The effectiveness of code review drops after around an hour. Set aside time throughout your day to coincide with breaks in order to not disrupt your flow.
+* Review at least part of the code, even if you can't do all of it.
```

Comment:
One thing I worry about are partial reviews -- that either block merges, or imply a thumbs up. 

If you can't review everything, how do you signal the author that you'll be back later, or that you need someone else to complete the review?